src/Saturn/Channels.fs

```diff
@@ -130,21 +87,34 @@ module Channels =
                         let! joinResult = channel.Join ctx
                         match joinResult with
                         | Ok ->
+                            let ctok = ctx.RequestAborted
                             let! webSocket = ctx.WebSockets.AcceptWebSocketAsync()
-                            let socketId = sockets.ConnectSocketToPath path webSocket
-
-                            let buffer : byte [] = Array.zeroCreate defaultBufferSize //It's buffer for just open message.
-                            let! echo = webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None)
-
-                            while not echo.CloseStatus.HasValue do
-                                let! (result, msg) = receiveMessageAsUTF8 CancellationToken.None webSocket
-                                let msg = serializer.Deserialize<Message> msg
-                                do! channel.HandleMessage(ctx, result, msg)
-                                ()
-
-                            do! channel.Terminate ctx
-                            sockets.DisconnectSocketForPath path socketId
-                            do! webSocket.CloseAsync(echo.CloseStatus.Value, echo.CloseStatusDescription, CancellationToken.None)
+                            let wrappedSocket = Socket.createFromWebSocket (Dataflow.DataflowBlockOptions()) webSocket // TODO: figure out what our datablock options should be
+                            let socketId = sockets.ConnectSocketToPath path wrappedSocket
+
+                            match! Socket.receiveMessageAsUTF8 wrappedSocket ctok with
+                            | Core.Ok _ ->
+                              while wrappedSocket.CloseStatus.IsSome do
+                                match! Socket.receiveMessageAsUTF8 wrappedSocket ctx.RequestAborted with
+                                | Core.Ok msg ->
+                                  let msg = serializer.Deserialize<Message> msg
+                                  do! channel.HandleMessage(ctx, msg)
+                                  ()
+                                | Core.Error exn ->
+                                  () // TODO: ?
```

Comment:
Usually if there's an exception the socket connection is killed so just grabbing the Aspnetcore logger and logging it is good enough